/*
 * Use Protocol Buffer version 3.
 */ 
syntax = "proto3";

/*
 * This data type carries information about recipient of the message.
 * Message itself is encrypted using symetric key.
 * Symetric key is encrypted using target's public key.
 * It is implementation of hybrid encryption.
 */ 
message TargetMessage
{
    // Recipient id.
    string targetId      = 1;
    // Symertic key encrypted with recipient's public key.
    bytes encSymetricKey = 2;
    // Data encrypted with symertic key.
    bytes encData        = 3;
}
